File: Submissions/1021_Remove_Outermost_Parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string S) {
        string result = "";
        int open = 0;
        
        for(int i=0; i<S.size(); ++i){
            if(S[i]=='(') {
                ++open;
                if(open>1) result += '(';
            } else if(S[i]==')') {
                --open;
                if(open>0) result += ')';
            }
        }
        
        return result;
    }
};

File: Kotlin/1021_Remove_Outermost_Parentheses.kt
class Solution {
    fun removeOuterParentheses(S: String): String {
        val result = StringBuilder()
        var open = 0
        
        for(c in S){
            if(c == '('){
                ++open
                if(open <= 1) continue
            }else{
                --open
                if(open <= 0) continue
            }
            result.append(c)
        }
        
        return result.toString()
    }
}